noir/noir-repo/compiler/noirc_evaluator/src/ssa/opt/die.rs

```diff
@@ -1009,7 +1074,7 @@ mod test {
         let ssa = Ssa::from_str(src).unwrap();
 
         // Even though these ACIR functions only have 1 block, we have not inlined and flattened anything yet.
-        let ssa = ssa.dead_instruction_elimination_inner(false);
+        let ssa = ssa.dead_instruction_elimination_inner(false, false);
```

Comment:
It is because this test that I added the flag, it looks like it was meant to be removed, and in the example `main` does get an array as input. I thought about making an exception for `main` (function with 0 ID), but I wasn't sure if this would be an important behaviour to keep.